edgedb/abstract.py

```diff
@@ -0,0 +1,63 @@
+import abc
+import typing
+
+from .datatypes import datatypes
+
+
+__all__ = ('Executor', 'AsyncIOExecutor')
+
+
+class ReadOnlyExecutor(abc.ABC):
+
+    @abc.abstractmethod
+    def query(self, query: str, *args, **kwargs) -> datatypes.Set:
+        ...
+
+    @abc.abstractmethod
+    def query_one(self, query: str, *args, **kwargs) -> typing.Any:
+        ...
+
+    @abc.abstractmethod
+    def query_json(self, query: str, *args, **kwargs) -> str:
+        ...
+
+    @abc.abstractmethod
+    def query_one_json(self, query: str, *args, **kwargs) -> str:
+        ...
+
+    # TODO(tailhook) add *args, **kwargs, when they are supported
+    @abc.abstractmethod
+    def execute(self, query: str) -> None:
+        ...
+
+
+class Executor(abc.ABC):
+    pass
```

Comment:
What is this class for? Can you add a docstring?

Reply:
Fixed inheritance and a docstring.